src/Razor/test/Microsoft.AspNetCore.Razor.ProjectEngineHost.Test/Serialization/GenerateJsonFiles.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+// Uncomment to easily generate new JSON files
+//#define GENERATE_JSON_FILES
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.ProjectSystem;
+using Microsoft.AspNetCore.Razor.Serialization.Json;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.CodeAnalysis.Razor.Logging;
+using Xunit;
+using Xunit.Abstractions;
+
+///////////////////////////////////////////////////////////////////////////////
+//
+// Note: The JSON files used for testing are very large. When making
+// significant changes to the JSON format for tag helpers or RazorProjectInfo, it
+// can be helpful to update the ObjectWriters first and the write new JSON files
+// before updating the ObjectReaders. This avoids having to make a series of
+// manual edits to the JSON resource files.
+//
+// 1. Update ObjectWriters to write the new JSON format.
+// 2. Uncomment the GENERATE_JSON_FILES #define above.
+// 3. Run the GenerateNewJsonFiles test below.
+// 4. Update ObjectReaders to read the new JSON format.
+// 5. Comment the GENERATE_JSON_FILES #define again.
+// 6. Run all of the tests in SerializerValidationTest to ensure that the
+//    new JSON files deserialize correctly.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+namespace Microsoft.AspNetCore.Razor.ProjectEngineHost.Test.Serialization;
+
+public class GenerateJsonFiles(ITestOutputHelper testOutput) : ToolingTestBase(testOutput)
+{
+#if GENERATE_JSON_FILES
+    internal static readonly bool ShouldGenerate = true;
+#else
+    internal static readonly bool ShouldGenerate = false;
+#endif
+
+    // This is to prevent you from accidentally checking in with GenerateJsonFiles = true
+    [Fact]
+    public void GenerateJsonFilesMustBeFalse()
+    {
+        Assert.False(ShouldGenerate, "GenerateJsonFiles should be set back to false before you check in!");
+    }
+
+    // This updates shared JSON files
+    [Theory]
+    [MemberData(nameof(JsonFiles))]
+    public void GenerateNewJsonFiles(JsonFile jsonFile)
+    {
+        if (!ShouldGenerate)
```

Comment:
Consider using `Skip = ShouldGenerate` in the attribute instead, that way the test is marked as skipped instead of appearing green in the test explorer.

Reply:
`ShouldGenerate` can't be used in that way because it's not a const. Like `GenerateBaselines.ShouldGenerate`, it's intentionally a static readonly field so that code that checks it doesn't get highlighted as dead code.

Reply:
Maybe I'm missing something, but there appears to only be 2 users of `GenerateJsonFiles.ShouldGenerate`, and they're both in this type. Why couldn't we just make it a const and use them as skipped conditions here?

Reply:
We could! I was expecting that `GenerateJsonFiles.ShouldGenerate` would eventually be used similarly to `GenerateBaselines.ShouldGenerate`, and then it made more sense to make them the same. But I guess it doesn't really matter. I'll change it.

Reply:
Note that the const/static readonly thing really doesn't matter anyway. Skip takes a string, not a bool. :smile: